client/src/main/java/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -742,33 +743,36 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
    |        END OF EVENT HANDLER                                |
    * ========================================================== */
 
-  def reserveSlots(
-    applicationId: String, shuffleId: Int, slots: WorkerResource): util.List[WorkerInfo] = {
-    val failed = new util.ArrayList[WorkerInfo]()
+  def reserveSlots(applicationId: String, shuffleId: Int,
+    slots: WorkerResource): ConcurrentSet[WorkerInfo] = {
 
-    slots.asScala.foreach { entry =>
+    val reserveSlotsFailedWorkers = new ConcurrentSet[WorkerInfo]
+
+    val parallelism = Array(slots.size(), RssConf.rpcMaxParallelism(conf),
```

Comment:
Currently RssConf.rpcMaxParallelism is used in server side and defaults to 1024. We'd better change the README and add the description of spark.rss.rpc.max.parallelism

Reply:
I add limit for the parallelism commitFiles, and split current rpc timeout config for different actions